src/chapters/more-on-functions/recursion-walkthrough.rst

```diff
@@ -0,0 +1,207 @@
+      //some code here
+   }
+
+We want ``combineEntries`` to repeat over and over again until the task is
+complete.
+
+How do we make this happen without using ``for`` or ``while``?
+
+.. index:: ! base case
+
+Identifying the Base Case
+--------------------------
+
+| ``for`` and ``while`` loops end when a particular condition evaluates to
+   ``false``. In the examples above, these conditions are
+| ``i < arr.length`` and ``arr.length > 0``, respectively.
+
+With recursion, we do not know how many times ``combineEntries`` must be
+called. To make sure the code stops at the proper time, we need to identify a
+condition that ends the process. This is called the **base case**, and it
+represents the simplest possible task for our function.
+
+``if`` the base case is ``true``, the recursion ends and the task is complete.
+``if`` the base case is ``false``, the function calls itself again.
+
+We check for the base case like this:
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      if (baseCase is true){
+         //solve last step
+         //end recursion
+      } else {
+         //solve next small step
+         //call combineEntries again
+      }
+   }
+
+For the joining task, the *base case* occurs when we pass in a one-element
+array (e.g. ``[ 'L' ]``). With no other elements to join together, the function
+just needs to return ``'L'``.
+
+Let's update ``combineEntries`` to check if the array contains only one item.
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      if (arrayName.length === 1){
+         return arrayName[0];
+      } else {
+         //solve next small step
+         //call combineEntries again
+      }
+   }
+
+``arrayName.length === 1`` sets up the condition for ending the recursion
+process. If it is ``true``, the single entry gets returned, and the function
+stops. Otherwise, ``combineEntries`` gets called again.
+
+Check Your Understanding
+-------------------------
+
+.. admonition:: Question
+
+   We can use recursion to remove all of the 'i' entries from the array
+   ``['One', 'i', 'c', 'X', 'i', 'i', 54]``.
+
+   Consider the code sample below, which declares the ``removeI`` function.
+
+   .. sourcecode:: js
+
+      function removeI(arr) {
+         if (baseCase is true){
+            //return final array
+            //end recursion
+         } else {
+            //remove one 'i' entry from array
+            //call removeI function again
+         }
+      };
+
+   Which TWO of the following work as a base case for the function? Feel free to
+   test the options in the repl.it to check your thinking.
+
+   #. ``!arr.includes('i')``
+   #. ``arr.includes('i')``
+   #. ``arr.indexOf('i')===-1``
+   #. ``arr.indexOf('i') !== -1``
+
+   Try it `here <https://repl.it/@launchcode/BaseCaseCC01>`__.
+
+.. admonition:: Question
+
+   The **factorial** of a number (n!) is the product of a positive, whole number and
+   all the integers below it.
```

Comment:
This definition isn't quite right, since integers can be negative, so "all integers below it" would include an infinite set of negative ints. The classic definition of factorial is purely symbolic, which avoids potential awkwardness with the math terms: `n! = 1 * 2 * 3*...* n-1 * n`.

Reply:
Google failed!  I wanted to be accurate, so I queried "definition of factorial", and this is what  lord Google told me.

I fixed the definition.